pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -246,7 +246,18 @@ func (s *Service) DeregisterInstanceFromAPIServerELB(i *infrav1.Instance) error
 	}
 
 	_, err = s.scope.ELB.DeregisterInstancesFromLoadBalancer(input)
-	return err
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case elb.ErrCodeAccessPointNotFoundException, elb.ErrCodeInvalidEndPointException:
+				// Ignoring LoadBalancerNotFound and InvalidInstance when deregistering
+				return nil
+			default:
+				return err
+			}
+		}
```

Comment:
Apologies for the late catch, but I just noticed this would swallow the error if for some reason the error returned is not an `awserr.Error`

```suggestion
		}
		return err
```

Reply:
gah - missed it too. Switching back to returning `err` there.